apps/backend/src/users/users.service.ts

```diff
@@ -41,4 +35,37 @@ export class UsersService {
 
     return users;
   }
+
+  async findOne(userId: number) {
+    const user = await this.usersRepository.findOneBy({ id: userId });
+
+    if (!user) {
+      throw new BadRequestException('User not found');
+    }
+
+    const currentUser = getCurrentUser();
+
+    const currentStatus = currentUser.status;
+    const targetStatus = user.status;
+    switch (currentStatus) {
+      case Status.RECRUITER:
+        if (targetStatus === Status.ADMIN) {
+          throw new BadRequestException('User not found');
+        }
+        break;
+      case Status.APPLICANT:
+        if (currentUser.userId !== user.userId) {
+          throw new BadRequestException('User not found');
+        }
+        break;
+      case Status.MEMBER:
+      case Status.ALUMNI:
+        if (currentUser.status === Status.APPLICANT) {
```

Comment:
🔧Should this be `targetStatus` instead of `currentUser.status` since members and alumni shouldn't be able to view the information of applicants as per the ticket specs? I can currently view applicant information as a member.